### PR TITLE
[OBSDEF-9231] giving permissions for default sa

### PR DESCRIPTION
**Pull request by itisevgeny**
## Purpose
OBSDEF-9231 Operator cannot provision in end user namespace

Matters when objstore is being deployed in end user namespace (objcontrolsvc at least needs these).
When it is being deployed in objscale ns permissions are coming from zookeeper-operator (which is definitely a bug and should be discussed and reworked later on).

## PR checklist
- [x] make test passed
- [ ] make build passed
- [x] helm install <chart> passed
- [ ] vSphere7 make package deployments passed
- [ ] helm upgrade <chart> passed
- [ ] helm test <release_name> passed
- [ ] Passed - https://asd-ecs-jenkins.isus.emc.com/jenkins/view/ECS-Flex/job/charts-custom-ci/

## Testing
https://asd-ecs-jenkins.isus.emc.com/jenkins/view/ECS-Flex/job/charts-custom-ci/230/

Charts:      28 passed, 28 total
Test Suites: 51 passed, 51 total
Tests:       245 passed, 245 total
Snapshot:    0 passed, 0 total
Time:        24.282357886s

### CODE REVIEW

**Review by antonbassin — APPROVED** [review]

File: ecs-cluster/templates/objectstore-rbac.yaml [hunk 1]
```diff
@@ -73,3 +73,50 @@ roleRef:
...
+    app.kubernetes.io/name: {{ .Release.Name }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/version: {{ .Chart.AppVersion }}
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name}}
+    product: objectscale
+subjects:
+  - kind: ServiceAccount
+    name: default
```
Comment (eddiepavkovic) [inline ecs-cluster/templates/objectstore-rbac.yaml:117]: is this temporary until objectcontrol gets their own service account.

Comment (itisevgeny) [reply ecs-cluster/templates/objectstore-rbac.yaml:117]: We need to agreed on some common solution. I'd better create a story for creating per-component accounts for all components with required permissions only. Then remove this change and remove binding from default sa in zookeeper operator.

**Review by tmarcu — COMMENTED** [review]
I'm a little worried about adding a rolebinding to the default service account that lets people get & **update** secrets, since every pod uses default unless another is specified. Can we make a provision RBAC and maybe update the operator to add the provision SA to the pods instead? I see this sticking around for a while unless someone takes on the following tasks like you described - creating per component account with required perms.

**Comment by itisevgeny** [comment]
> I'm a little worried about adding a rolebinding to the default service account that lets people get & **update** secrets, since every pod uses default unless another is specified. Can we make a provision RBAC and maybe update the operator to add the provision SA to the pods instead? I see this sticking around for a while unless someone takes on the following tasks like you described - creating per component account with required perms.

@tmarcu I have created new story for dealing with service accounts. Can we merge it as is? I'd prefer to make modifications in operator spec under OBSDEF-9688.

**Comment by antonbassin** [comment]
> > I'm a little worried about adding a rolebinding to the default service account that lets people get & **update** secrets, since every pod uses default unless another is specified. Can we make a provision RBAC and maybe update the operator to add the provision SA to the pods instead? I see this sticking around for a while unless someone takes on the following tasks like you described - creating per component account with required perms.
> 
> @tmarcu I have created new story for dealing with service accounts. Can we merge it as is? I'd prefer to make modifications in operator spec under OBSDEF-9688.

@itisevgeny let's merge it now, cause it blocks the further testing

**Review by BassinD — APPROVED** [review]

**Pull request merged**